### PR TITLE
Add posibility to load upsell/interstitial

### PR DESCRIPTION
**Pull request by MariuszTasak**
Commit introduces:

new file:   src/Nexway/SetupManager/Util/Processor/Action/Interstitialpage/Create.php
new file:   src/Nexway/SetupManager/Util/Processor/Action/Interstitialpage/README.md
new file:   src/Nexway/SetupManager/Util/Processor/Action/Upsell/Create.php

modified:   src/Nexway/SetupManager/Util/Helper/Data.php
modified:   src/Nexway/SetupManager/Util/Helper/Parser.php
new file:   src/Nexway/SetupManager/Util/Processor/Action/Upsell/README.md

modified:   composer.json

### CODE REVIEW

File: src/Nexway/SetupManager/Util/Processor/Action/Upsell/Create.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    /**
+     * @return bool
+     */
+    protected function _create()
+    {
+        $this->getParameters()->setModel('nexway_upsell/upsell');
+
+        $data = $this->getParameters()->getData(self::F_DATA);
+
+        $data['upsell_products'] = $data['source_product_id'] . '::' . $data['target_product_id'];
```
Comment (marcinrogacki) [inline src/Nexway/SetupManager/Util/Processor/Action/Upsell/Create.php]: :+1: for hide here (instead in config files) a way how products are paired

File: src/Nexway/SetupManager/Util/Processor/Action/Upsell/Create.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+        /** @var Nexway_Upsell_Model_Options $optionsModel */
+        $optionsModel   = \Mage::getModel('nexway_upsell/options');
+
+        $data = $optionsModel->parseData($data);
+        $upsellModel->setData($data);
+        $upsellModel->save();
+
+        $upsellId = $upsellModel->getId();
+        $upsellModel->setUpsellId($upsellId);
```
Comment (marcinrogacki) [inline src/Nexway/SetupManager/Util/Processor/Action/Upsell/Create.php]: Get upsell id to set it back in `$upsellModel`?

File: src/Nexway/SetupManager/Util/Processor/Action/Upsell/Create.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+        $optionsModel->setUpsellId($upsellId);
+        $optionsModel->removeOptions($upsellId);
```
Comment (marcinrogacki) [inline src/Nexway/SetupManager/Util/Processor/Action/Upsell/Create.php]: I don't understand but I guess it some magic of Upsell module.

File: src/Nexway/SetupManager/Util/Processor/Action/Upsell/Create.php [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        $optionsModel->saveIncomingData();
```
Comment (marcinrogacki) [inline src/Nexway/SetupManager/Util/Processor/Action/Upsell/Create.php]: I don't understand but I guess it some magic of Upsell module.

File: src/Nexway/SetupManager/Util/Processor/Action/Upsell/Create.php [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        }
+
+        if (!$upsellModel->validateEntity()) {
+            $upsellModel->setStatus(
+                \Nexway_Upsell_Model_Source_Status::UPSELL_MISCONFIGURED
+            );
+            /** form data validation requires actual upsell id */
+            $upsellModel->setUpsellId($upsellModel->getId());
+            $upsellModel->save();
+        }
```
Comment (marcinrogacki) [inline src/Nexway/SetupManager/Util/Processor/Action/Upsell/Create.php]: :+1: for not forget about validation

**Comment by marcinrogacki** [comment]
review :+1: just one or two questions to check

File: src/Nexway/SetupManager/Util/Processor/Action/Upsell/Create.php (hunk 2 above)
Comment (MariuszTasak) [inline src/Nexway/SetupManager/Util/Processor/Action/Upsell/Create.php]: I don't know why, but upsell model requires upsell_is field for validation. I simplified logic from controller, but i hadn't time to refactor whole validation and save process.

File: src/Nexway/SetupManager/Util/Processor/Action/Upsell/Create.php (hunk 4 above)
Comment (MariuszTasak) [inline src/Nexway/SetupManager/Util/Processor/Action/Upsell/Create.php]: yes

**Comment by pawlik** [comment]
:+1:

**Pull request merged**